independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java

```diff
@@ -617,9 +623,12 @@ private void processDecorator(DecoratorInfo decorator, BeanInfo bean, Type provi
         }
         ClassCreator delegateSubclass = delegateSubclassBuilder.build();
 
-        Map<MethodDescriptor, DecoratorInfo> nextDecorators = bean.getNextDecorators(decorator);
-        Collection<DecoratorInfo> nextDecoratorsValues = nextDecorators.values();
-        List<DecoratorInfo> decoratorParameters = new ArrayList<>(new HashSet<>(nextDecoratorsValues));
+        Map<MethodDescriptor, DecoratorMethod> nextDecorators = bean.getNextDecorators(decorator);
+        Set<DecoratorInfo> nextDecoratorsValues = new HashSet<>();
```

Comment:
The set here is used to eliminate possible duplicates? Just for my understanding.